article_store/articles/urls.py

```diff
@@ -0,0 +1,22 @@
+from django.urls import path, include
+
+from rest_framework import routers
+
+from articles.api import (
+    ArticleViewSet,
+    ArticleXMLAPIView,
+    ArticleListAPIView,
+)
+
+
+router_v1 = routers.DefaultRouter()
+router_v1.register('articles', ArticleViewSet, base_name='article')
+
+
+urlpatterns = [
+    path('/api/v1/', include((router_v1.urls, 'api_v1'))),
+    path('/<str:article_id>', ArticleXMLAPIView.as_view(), name='article-xml'),
+    path('/<str:article_id>/<str:version>', ArticleXMLAPIView.as_view(), name='article-version'),
+    path('/<str:article_id>/<str:version>/<str:part>', ArticleXMLAPIView.as_view(), name='article-content'),
+    path('', ArticleListAPIView.as_view(), name='article-list'),
+]
```

Comment:
I get some warnings:
```
article-store_1    | WARNINGS:
article-store_1    | ?: (urls.W002) Your URL pattern '/<str:article_id>' [name='article-xml'] has a route beginning with a '/'. Remove this slash as it is unnecessary. If this pattern is targeted in an include(), ensure the include() pattern has a trailing '/'.
article-store_1    | ?: (urls.W002) Your URL pattern '/<str:article_id>/<str:version>' [name='article-version'] has a route beginning with a '/'. Remove this slash as it is unnecessary. If this pattern is targeted in an include(), ensure the include() pattern has a trailing '/'
```

Reply:
These are because of the removal of the trailing `/` (which Django likes) on the `/articles` endpoint. Which means I have to include it in subsequent path matches.

I will change a setting which should remove this check as its not needed.